fhir-bucket/src/main/java/com/ibm/fhir/bucket/client/FhirClient.java

```diff
@@ -0,0 +1,504 @@
+            logger.severe("Error while executing the PUT request. "+e);
+            logger.warning("PUT URL: "+target+"\nRequest Body: "+body);
+        } catch (IOException e) {
+            logger.severe("Error while executing the PUT request. "+e);
+            logger.warning("PUT URL: "+target+"\nRequest Body: "+body);
+        }
+        
+        return null;
+    }
+    
+    public void shutdown() {
+        if (client != null) {
+            try {
+                connManager.shutdown();
+                client.close();
+            } catch (IOException e) {
+                throw new IllegalStateException("Unable to shutdown HTTP clients and Connection Manager successfully. ", e);
+            }
+        }
+        
+    }
+
+    /**
+     * Get statistics from the internal HTTP connection manager
+     * @return
+     */
+    public PoolStats getPoolInformation() {
+        if (connManager != null) {
+            return connManager.getTotalStats();
+        } 
+        
+        return null;
+    }
+
+    /**
+     * Construct a FhirServerResponse from the FHIR server {@link HttpResponse}
+     * @param response
+     * @param startTime
+     * @return
+     */
+    private FhirServerResponse buildResponse(HttpResponse response, long startTime, boolean processResponseEntity) {
+        FhirServerResponse sr = new FhirServerResponse();
+        
+        int status = response.getStatusLine().getStatusCode();
+        sr.setStatusCode(status);
+        sr.setStatusMessage(response.getStatusLine().getReasonPhrase());
+
+        HttpEntity entity = response.getEntity();
+        try {
+            if (status == HttpStatus.SC_OK || status == HttpStatus.SC_CREATED) {
+                if (processResponseEntity) {
+                    processEntity(sr, entity);
+                } else if (response.getFirstHeader("Location") != null) {
+                    // Single resource case, no response body, just the URL returned in the Location header
+                    sr.setLocationHeader(response.getFirstHeader("Location").getValue());
+                } else {
+                    logger.warning("No body or Location header in response");
+                }
+            } else if (status == HttpStatus.SC_BAD_REQUEST) {
+                processOperationalOutcome(sr, entity);
+            } else {
+                logger.warning("Unexpected server response: " + status + " " + response.getStatusLine().getReasonPhrase());
+            }
+        } finally {
+            consume(entity);
+            long endTime = System.nanoTime();
+            sr.setResponseTime((int)(endTime-startTime));
+        }
+        
+            // Last-Modified: 2018-11-26T05:07:00.954Z
+            // TODO
+//            Header lastModifiedHeader = response.getFirstHeader("Last-Modified");
+//            if (lastModifiedHeader != null) {
+//                sr.setLastModified(TimeUtil.getFhirTime(lastModifiedHeader.getValue()));
```

Comment:
I think we have a setting that we can flip between FHIR and the RFC lastmodified.  In case you hit an issue with the format.

Reply:
I'm not using this at the moment, but noted for when I add this. Hopefully we have some utility code which already handles this, if not I create something.